src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -415,11 +415,20 @@ def _calculate_mask_iou(mask1: Tensor, mask2: Tensor) -> Tensor:
             overlapped_label = []
             overlapped_other_label = []
             for (im, mask), (jm, other_mask) in product(enumerate(masks), enumerate(other_masks)):
-                if _calculate_mask_iou(mask, other_mask) > threshold_iou:
+                _mask_iou = _calculate_mask_iou(mask, other_mask)
+                if _mask_iou > threshold_iou:
+                    # compare overlapped regions between different labels and filter out the lower score
                     if used_points[label][im][2] > used_points[other_label][jm][2]:
                         overlapped_other_label.append(jm)
                     else:
                         overlapped_label.append(im)
+                elif _mask_iou > 0:
+                    # refine the slightly overlapping region
+                    overlapped_coords = torch.where(torch.logical_and(mask, other_mask))
```

Comment:
Reuse `torch.logical_and(mask, other_mask)` at L404 not to compute it twice

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/3040/commits/d248f81cf9ac949984d986ad1730b388a57131cf.